tests/test_webhook.py

```diff
@@ -527,5 +529,96 @@ def test_handler(self):
 
 
+def wrap(func):
+    def wrapper(*args):
+        if PY3:
+            arg_spec = inspect.getfullargspec(func)
+        else:
+            arg_spec = inspect.getargspec(func)
+        return func(*args[0:len(arg_spec.args)])
+    return wrapper
+
+
+class TestWebhookHandlerWithWrappedFunction(unittest.TestCase):
+    def setUp(self):
+        self.handler = WebhookHandler('channel_secret')
+
+        @self.handler.add(MessageEvent, message=TextMessage)
+        @wrap
+        def message_text(event, destination):
+            self.assertEqual('message', event.type)
+            self.assertEqual('text', event.message.type)
+            self.assertEqual('U123', destination)
+
+        @self.handler.add(MessageEvent,
+                          message=(ImageMessage, VideoMessage, AudioMessage))
+        @wrap
+        def message_content(event):
+            self.assertEqual('message', event.type)
+            self.assertIn(
+                event.message.type,
+                ['image', 'video', 'audio']
+            )
+
+        @self.handler.add(MessageEvent, message=StickerMessage)
+        @wrap
+        def message_sticker(event):
+            self.assertEqual('message', event.type)
+            self.assertEqual('sticker', event.message.type)
+
+        @self.handler.add(MessageEvent)
+        @wrap
+        def message(event):
+            self.assertEqual('message', event.type)
+            self.assertNotIn(
+                event.message.type,
+                ['text', 'image', 'video', 'audio', 'sticker']
+            )
+
+        @self.handler.add(FollowEvent)
+        @wrap
+        def follow(event, destination):
+            self.assertEqual('follow', event.type)
+            self.assertEqual('U123', destination)
+
+        @self.handler.add(JoinEvent)
+        @wrap
+        def join(event):
+            self.assertEqual('join', event.type)
+
+        @self.handler.add(PostbackEvent)
+        @wrap
+        def postback(event):
+            self.assertEqual('postback', event.type)
+
+        @self.handler.add(BeaconEvent)
+        @wrap
+        def beacon(event):
+            self.assertEqual('beacon', event.type)
+
+        @self.handler.add(AccountLinkEvent)
+        @wrap
+        def account_link(event):
+            self.assertEqual('accountLink', event.type)
+
+        @self.handler.default()
+        def default(event):
+            self.assertNotIn(
+                event.type,
+                ['message', 'follow', 'join', 'postback', 'beacon', 'accountLink']
+            )
+
+    def test_handler_with_wrapped_function(self):
+        file_dir = os.path.dirname(__file__)
+        webhook_sample_json_path = os.path.join(file_dir, 'text', 'webhook.json')
+        with open(webhook_sample_json_path) as fp:
+            body = fp.read()
+
+        # mock
+        self.handler.parser.signature_validator.validate = lambda a, b: True
+
+        self.handler.handle(body, 'signature')
+
+
```

Comment:
[imo]
Test seems good, but it seems redundant to make`TestWebhookHandlerWithWrappedFunction`.

If we separate the part which invokes function from WebhookHandler.handle like following and test it, the test will be more clear and simple I think.

```python
    def __invoke_func(self, func, event, payload):
        has_varargs, args_count = self.__get_args_count(func)
        if has_varargs or args_count >= 2:
            func(event, payload.destination)
        elif args_count == 1:
            func(event)
        else:
            func()
```